src/main/java/jcmp/DocCompare.java

```diff
@@ -394,6 +393,9 @@ public int hashCode() {
 
         @Override
         public boolean equals(Object x) {
+            if (!(x instanceof DefaultIdentity)) {
```

Comment:
My point was that for most executions try-catch has no overhead, whereas
for all executions there will be overhead for checking instanceof. No need
for null check, instanceof implies non-null.

On Tue, Mar 29, 2016 at 11:01 AM, Naveen Malik <notifications@github.com>
wrote:

> In src/main/java/jcmp/DocCompare.java
> <https://github.com/bserdar/jiff/pull/1#discussion_r57759588>:
>
> > @@ -394,6 +393,9 @@ public int hashCode() {
> >
> >          @Override
> >          public boolean equals(Object x) {
> > +            if (!(x instanceof DefaultIdentity)) {
>
> Sure, but isn't an if going to be less expensive than a class cast
> exception being created, thrown, then caught? Could extend the if to check
> for null then get rid of the try/catch completely.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/bserdar/jiff/pull/1/files/4efbd8f32d48864623357249c53f773bdd917044#r57759588>
>